packager/media/formats/mp4/mp4_media_parser.cc

```diff
@@ -604,7 +604,15 @@ bool MP4MediaParser::ParseMoov(BoxReader* reader) {
             LOG(ERROR) << "Failed to parse av1c.";
             return false;
           }
-          codec_string = av1_config.GetCodecString();
+          // Generate the full codec string if the colr atom is present.
+          if (entry.colr.color_parameter_type == "nclx") {
+            codec_string = av1_config.GetCodecString(
+                entry.colr.color_primaries, entry.colr.transfer_characteristics,
+                entry.colr.matrix_coefficients,
+                entry.colr.video_full_range_flag);
```

Comment:
I think transfer_characteristics should be set to entry.colr.transfer_characteristics after this line. 
It is used to create VideoStreamInfo on line 720 with which MediaInfo is created in HlsNotifyMuxerListener::OnMediaStart and MediaPlaylist::GetVideoRange() uses this MediaInfo to add the VIDEO-RANGE tag.

Reply:
Do you think it makes sense to add an E2E test (https://github.com/shaka-project/shaka-packager/blob/main/packager/app/test/packager_test.py)?